src/Microsoft.AspNet.Mvc.Razor/IRazorViewEngine.cs

```diff
@@ -11,12 +11,38 @@ namespace Microsoft.AspNet.Mvc.Razor
     public interface IRazorViewEngine : IViewEngine
     {
         /// <summary>
-        /// Finds a <see cref="IRazorPage"/> using the same view discovery semantics used in
-        /// <see cref="IViewEngine.FindPartialView(ActionContext, string)"/>.
+        /// Finds the page with the given <paramref name="pageName"/> using view locations and information from the
+        /// <paramref name="context"/>.
         /// </summary>
         /// <param name="context">The <see cref="ActionContext"/>.</param>
-        /// <param name="page">The name or full path to the view.</param>
-        /// <returns>A result representing the result of locating the <see cref="IRazorPage"/>.</returns>
-        RazorPageResult FindPage(ActionContext context, string page);
+        /// <param name="pageName">The name of the page.</param>
+        /// <param name="isMainPage">Determines if the page being found is the main page for an action.</param>
+        /// <returns>The <see cref="RazorPageResult"/> of locating the page.</returns>
+        /// <remarks><seealso cref="IViewEngine.FindView"/>.</remarks>
+        RazorPageResult FindPage(ActionContext context, string pageName, bool isMainPage);
```

Comment:
But there isn't a property `IsMainPage` on RazorPage (and probably shouldn't be one). We shouldn't need this parameter.